feat/71_매너온도 로직

dlrms6172 opened the pull request.

leeguen committed:
feat/71_매너온도 로직

leeguen committed:
Merge branch 'dev' of https://github.com/dlrms6172/cucumber_market_backend into feat/71_manners_temperature

jeongmallro commented on src/main/java/com/cucumber/market/api/dto/item/ItemDto.java:
```diff
@@ -53,6 +54,8 @@ public static class modifyItemStatusDto {
         @NotNull
         private ItemStatus itemStatus;
         private Integer clientId;
+        private BigDecimal incSellerMannersTemperature;
```
어차피 매너 온도는 소숫점 첫번째 자리까지만 보여주니까 double 을 사용했어도 괜찮았을 것 같습니다!

dlrms6172 replied:
말씀주신 double을 사용할 수 도 있겠으나, 좀 더 정확한 계산을 위해 BigDecimal을 사용했습니다.
double은 부동소수점 숫자이므로 계산할때 미세한 오차가 발생할 수 있습니다.
BigDecimal은 고정소수점 숫자이기 때문에 소수점 첫번째 자리계산과 표현하기에는 BigDecimal이 더 적합하다고 생각합니다.

jeongmallro commented on src/main/java/com/cucumber/market/api/service/item/ItemService.java:
```diff
@@ -73,6 +74,31 @@ public Map modifyItemStatus(Integer memberId, Integer itemId, ItemDto.modifyItem
+                itemDto.setIncSellerMannersTemperature(sellerMannersTemperature); //판매자:증가된 매너온도 셋팅
+                userMapper.updateSellerMannersTemperature(memberId, itemDto); //판매자:매너온도 업데이트
+
+                itemDto.setIncBuyerMannersTemperature(sellerMannersTemperature); //구매자:증가된 매너온도 셋팅
```
입력값으로 buyerMannersTemperature 가 들어가야 하는데 잘못 쓰신 것 같습니다.

dlrms6172 replied:
아 그렇네요 수정하겠습니다. 감사합니다!

jeongmallro commented on src/main/resources/sqlmapper/mappers/UserMapper.xml:
```diff
@@ -52,4 +52,34 @@
+        update cucumber.member
+        set manners_temperature = #{dto.incBuyerMannersTemperature}
+        where 1=1
+          and member_id = #{dto.clientId}
```
여기서 dto.clientId 값으로 받을 거면 UserMapper 에서 updateBuyerMannersTemperature 의 매개변수로 clientId 는 받을 필요 없을 것 같습니다!

dlrms6172 replied:
그러네요, 수정하겠습니다, 감사합니다!

jeongmallro commented on src/main/resources/sqlmapper/mappers/UserMapper.xml:
```diff
@@ -52,4 +52,34 @@
+            manners_temperature
+        from cucumber.member
+        where 1=1
+          and member_id = #{dto.clientId}
```
여기 그냥 clientId 아닌가요? 확인 부탁드립니다!

dlrms6172 replied:
위에 내용과 마찬가지 내용이네요. 수정하겠습니다!

jeongmallro commented on src/main/java/com/cucumber/market/api/service/item/ItemService.java:
```diff
@@ -73,6 +74,31 @@ public Map modifyItemStatus(Integer memberId, Integer itemId, ItemDto.modifyItem
                 itemMapper.insertBuyerReview(itemId, itemDto.getClientId(), null);  //구매자 후기 테이블에 등록
                 itemMapper.insertSellerReview(itemId, null);  //판매자 후기 테이블에 등록
+
+                BigDecimal maxMannersTemperature = new BigDecimal("99");
```
매너온도 부분 로직이 길어서 따로 메서드로 빼도 괜찮을 것 같습니다.

jeongmallro commented on src/main/java/com/cucumber/market/api/dto/item/ItemDto.java:
```diff
@@ -8,6 +8,7 @@
+import java.math.BigDecimal;
 import java.time.LocalDateTime;
 
 public class ItemDto {
```
매너온도 필드 추가 schema.sql 파일에도 부탁드립니다~

dlrms6172 replied:
넵!

jeongmallro reviewed (commented):
1차 리뷰입니다!

dlrms6172 committed:
1.buyerMannersTemperature 잘못 기재 수정
2.clientId 매개변수 수정
3.매너온도 로직 따로 메서드로 작성
4.schema.sql의 member table에 manner_temperature 추가

jeongmallro reviewed (approved):
확인했습니다!
